### PR TITLE
v3rpc, etcdctl: snapshot integrity hash

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: etcdctl/ctlv3/command/snapshot_command.go [hunk 1]
```diff
@@ -261,18 +263,53 @@ func makeDB(snapdir, dbfile string) {
 	}
 	defer f.Close()
 
+	// get snapshot integrity hash
+	if _, err := f.Seek(-sha256.Size, os.SEEK_END); err != nil {
```
Comment (xiang90) [inline etcdctl/ctlv3/command/snapshot_command.go]: probably we want to add a flag called  "no-hash-check"? some people might want to do local snapshot copy. then they do not have a hash attached. and then the integrity checking is on them.

Comment (heyitsanthony) [inline etcdctl/ctlv3/command/snapshot_command.go]: ok flag added

File: etcdctl/ctlv3/command/snapshot_command.go [hunk 2]
```diff
@@ -94,6 +97,7 @@ func NewSnapshotRestoreCommand() *cobra.Command {
 	cmd.Flags().StringVar(&restoreClusterToken, "initial-cluster-token", "etcd-cluster", "Initial cluster token for the etcd cluster during restore bootstrap.")
 	cmd.Flags().StringVar(&restorePeerURLs, "initial-advertise-peer-urls", defaultInitialAdvertisePeerURLs, "List of this member's peer URLs to advertise to the rest of the cluster.")
 	cmd.Flags().StringVar(&restoreName, "name", defaultName, "Human-readable name for this member.")
+	cmd.Flags().BoolVar(&skipHashCheck, "skip-hash-check", false, "Ignore snapshot integrity hash value.")
```
Comment (xiang90) [inline etcdctl/ctlv3/command/snapshot_command.go]: Ignore snapshot integrity hash value (this is required when the snapshot generated comes from Maintenance Service)

Comment (xiang90) [inline etcdctl/ctlv3/command/snapshot_command.go]: also we need to change the recover doc?

Comment (xiang90) [inline etcdctl/ctlv3/command/snapshot_command.go]: https://github.com/coreos/etcd/blob/master/Documentation/op-guide/recovery.md#snapshotting-the-keyspace

If user does the copy thing mentioned here, they should skip-hash-check.

Comment (heyitsanthony) [inline etcdctl/ctlv3/command/snapshot_command.go]: ok all fixed up

**Comment by xiang90** [comment]
LGTM

**Pull request merged**